### PR TITLE
{lib}[intel/2018a] TensorFlow v1.6.0 w/ Python 3.6.4

### PR DESCRIPTION
**Pull request by smoors**
depends on ~~#6043~~

### CODE REVIEW

**Comment by mstud** [comment]
I think you forgot to add the wheel .eb for intel/2018a

**Comment by boegel** [comment]
@smoors I tried this myself while working on the `foss/2018a`, and the build failed for me (didn't have time to figure out what was wrong).
Does this work for you?

**Comment by smoors** [comment]
Of course, otherwise I would not make a PR ;)
I had to add another patch for intel, see at the end of TensorFlow-1.6.0_icc-fixes.patch.

**Comment by smoors** [comment]
@mstud yes, I made a separate PR for that (and forgot to mention it).

**Comment by boegel** [comment]
@smoors Are you referring to the `fix error 'invalid narrowing conversion from "unsigned long" to "int"' via explicit type casting` part of the patch?

We should maybe make this a separate patch? The large patch can be auto-generated, but if more patching is needed, we should keep it separate to update for future TF versions more easily?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2587.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/1d74cfa3154b774103077444da337be5 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2091.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/2f0229180095313fc8e58a20c9ced283 for a full test report.

**Comment by boegel** [comment]
Ignore the failing test reports, need patch that was included in #5951.

This is a bug in `--from-pr` I should try and get fixed...

File: easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.6.0-intel-2018a-Python-3.6.4.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+homepage = 'https://www.tensorflow.org/'
+description = "An open-source software library for Machine Intelligence"
+
+toolchain = {'name': 'intel', 'version': '2018a'}
+
+source_urls = ['https://github.com/tensorflow/tensorflow/archive/']
+sources = ['v%(version)s.tar.gz']
+patches = [
+    'TensorFlow-%(version)s_swig-env.patch',
```
Comment (boegel) [inline easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.6.0-intel-2018a-Python-3.6.4.eb]: @smoors Actually, no, seems like you forgot to include this patch @smoors?

Why not use the existing patch for TF 1.5.0 which still applies, which is what we also did in #5951.

Comment (smoors) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.6.0-intel-2018a-Python-3.6.4.eb]: Ah that patch is actually part of PR #6016. It is the same as the 1.5.0 version but with different line numbers. I do not know if eb is able to handle this.

Comment (boegel) [reply easybuild/easyconfigs/t/TensorFlow/TensorFlow-1.6.0-intel-2018a-Python-3.6.4.eb]: @smoors As long as `patch` is able to apply the patch file (it's smart enough to deal with slightly shifted patches), EB is happy too.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2055.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/5b3128231ba713b82a7d5deb01b86f82 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2694.swalot.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/f4ac417a61136c2f9d82c6ecf2b46a43 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @smoors!

**Pull request merged**